tests/jest/browser-locale/index.test.js

```diff
@@ -0,0 +1,150 @@
+import getBrowserLocale, { isSupported, filterLanguages } from '../../../src/browser-locale';
+
+describe('browser-locale/index', () => {
+  describe('exports', () => {
+    it('should export getBrowserLocale', () => {
+      expect(getBrowserLocale).toBeDefined();
+    });
+
+    it('should export isSupported', () => {
+      expect(isSupported).toBeDefined();
+    });
+
+    it('should export filterLanguages', () => {
+      expect(filterLanguages).toBeDefined();
+    });
+  });
+
+  describe('isSupported', () => {
+    it('should return false if TERRA_AGGREGATED_LOCALES is undefined', () => {
+      expect(isSupported('en')).toBe(false);
+    });
+
+    it('should return false if the locale is not within TERRA_AGGREGATED_LOCALES', () => {
+      global.TERRA_AGGREGATED_LOCALES = ['en', 'en-US'];
+
+      expect(isSupported('es')).toBe(false);
+    });
+
+    it('should return false if the locale is not provided', () => {
+      global.TERRA_AGGREGATED_LOCALES = ['en', 'en-US'];
+
+      expect(isSupported()).toBe(false);
+    });
+
+    it('should return false if TERRA_AGGREGATED_LOCALES is not an array', () => {
+      global.TERRA_AGGREGATED_LOCALES = 'en';
+
+      expect(isSupported('en')).toBe(false);
+    });
+
+    it('should return true if the locale is within TERRA_AGGREGATED_LOCALES', () => {
+      global.TERRA_AGGREGATED_LOCALES = ['es', 'en', 'en-US'];
+
+      expect(isSupported('en')).toBe(true);
+    });
+  });
+
+  describe('filterLanguages', () => {
+    beforeEach(() => {
+      Object.defineProperty(navigator, 'languages', { get() { return undefined; }, configurable: true });
+    });
+
+    afterEach(() => {
+      Object.defineProperty(navigator, 'languages', { get() { return undefined; }, configurable: true });
```

Comment:
curious - what's the reason we need to reset `navigator` with afterEach? it seems like the beforeEach would be enough?

Reply:
The intention is to be explicit about cleaning up any mock overrides. Without the afterEach cleanup the last test would leave the mocking active for any tests run there after in a different describe block. We could potentially change this to an afterAll.